clients/src/main/java/org/apache/kafka/clients/producer/KafkaProducer.java

```diff
@@ -1351,6 +1364,7 @@ private void close(Duration timeout, boolean swallowException) {
                         log.error("Interrupted while joining ioThread", t);
                     }
                 }
+                clientTelemetryReporter.ifPresent(reporter -> reporter.initiateClose(timeoutMs));
```

Comment:
Timeout handling is not correct. There are two operations both using the value of `timeoutMs`.